src/Symfony/Component/Console/DependencyInjection/AddConsoleCommandPass.php

```diff
@@ -103,16 +105,12 @@ public function process(ContainerBuilder $container): void
                 $definition->addMethodCall('setHidden', [true]);
             }
 
-            if (!$description) {
-                if (!$r = $container->getReflectionClass($class)) {
-                    throw new InvalidArgumentException(\sprintf('Class "%s" used for service "%s" cannot be found.', $class, $id));
-                }
-                if (!$r->isSubclassOf(Command::class)) {
-                    throw new InvalidArgumentException(\sprintf('The service "%s" tagged "%s" must be a subclass of "%s".', $id, 'console.command', Command::class));
-                }
-                $description = str_replace('%', '%%', $class::getDefaultDescription() ?? '');
+            if ($help && isset($invokableRef)) {
```

Comment:
I think it's worth adding a comment. The code is not that explicit.
```suggestion
            // If the command service extends the Command class, then there's no need to set help,
            // the Command class reads the AsCommand attribute itself.
            if ($help && isset($invokableRef)) {
```